sljit_src/sljitNativeRISCV_common.c

```diff
@@ -1357,33 +1417,67 @@ static SLJIT_INLINE sljit_s32 emit_single_op(struct sljit_compiler *compiler, sl
 #endif /* SLJIT_CONFIG_RISCV_64 */
 
 	case SLJIT_CLZ:
+#if defined __riscv_zbb
+		return push_inst(compiler, CLZ | WORD | RD(dst) | RS1(src2));
+#endif /* __riscv_zbb */
 	case SLJIT_CTZ:
+#if defined __riscv_zbb
+		return push_inst(compiler, CTZ | WORD | RD(dst) | RS1(src2));
+#else /* !__riscv_zbb */
 		SLJIT_ASSERT(src1 == TMP_ZERO && !(flags & SRC2_IMM));
 		return emit_clz_ctz(compiler, op, dst, src2);
+#endif /* __riscv_zbb */
 
 	case SLJIT_REV:
+#if defined __riscv_zbb
+		SLJIT_ASSERT(src1 == TMP_ZERO && !(flags & SRC2_IMM));
+		FAIL_IF(push_inst(compiler, REV8 | RD(dst) | RS1(src2)));
+#if defined SLJIT_CONFIG_RISCV_64 && SLJIT_CONFIG_RISCV_64
+		if (op & SLJIT_32)
+			return push_inst(compiler, SRAI | RD(dst) | RS1(dst) | IMM_I(32));
+		return SLJIT_SUCCESS;
+#else /* !SLJIT_CONFIG_RISCV_64 */
+		return SLJIT_SUCCESS;
+#endif /* SLJIT_CONFIG_RISCV_64 */
+#endif /* __riscv_zbb */
 	case SLJIT_REV_S32:
-#if (defined SLJIT_CONFIG_RISCV_32 && SLJIT_CONFIG_RISCV_32)
+#if ((defined SLJIT_CONFIG_RISCV_32 && SLJIT_CONFIG_RISCV_32) || defined __riscv_zbb)
 	case SLJIT_REV_U32:
-#endif /* SLJIT_CONFIG_RISCV_32 */
+#endif /* SLJIT_CONFIG_RISCV_32 || __riscv_zbb */
 		SLJIT_ASSERT(src1 == TMP_ZERO && !(flags & SRC2_IMM));
+#if defined __riscv_zbb
+		FAIL_IF(push_inst(compiler, REV8 | RD(dst) | RS1(src2)));
+#if defined SLJIT_CONFIG_RISCV_64 && SLJIT_CONFIG_RISCV_64
+		return push_inst(compiler, (GET_OPCODE(op) == SLJIT_REV_U32 ? SRLI : SRAI )| RD(dst) | RS1(dst) | IMM_I(32));
+#else /* !SLJIT_CONFIG_RISCV_64 */
+		return SLJIT_SUCCESS;
+#endif /* SLJIT_CONFIG_RISCV_64 */
+#else /* !__riscv_zbb */
 		return emit_rev(compiler, op, dst, src2);
-
+#endif /* __riscv_zbb */
 	case SLJIT_REV_U16:
 	case SLJIT_REV_S16:
 		SLJIT_ASSERT(src1 == TMP_ZERO && !(flags & SRC2_IMM));
+#if defined __riscv_zbb
+		FAIL_IF(push_inst(compiler, REV8 | RD(dst) | RS1(src2)));
+#if defined SLJIT_CONFIG_RISCV_64 && SLJIT_CONFIG_RISCV_64
+		return push_inst(compiler, (GET_OPCODE(op) == SLJIT_REV_U16 ? SRLI : SRAI )| RD(dst) | RS1(dst) | IMM_I(48));
+#else /* !SLJIT_CONFIG_RISCV_64 */
+		return push_inst(compiler, (GET_OPCODE(op) == SLJIT_REV_U16 ? SRLI : SRAI) | RD(dst) | RS1(dst) | IMM_I(16));
+#endif /* SLJIT_CONFIG_RISCV_64 */
+#else /* !__riscv_zbb */
 		return emit_rev16(compiler, op, dst, src2);
```

Comment:
this function shouldn't be needed for !__riscv_zbb, why is not #ifdef out like `emit_clz_ctz`?

Reply:
Hm, the compiler should complain about that.

Reply:
I think we do a lot of `static` functions with gcc that might just result on it not triggering, did you try `clang`?

Reply:
the `#ifdef` that I originally though was only for `emit_clz_ctz` actually covers all 3 functions, so this is a non issue.  should had pulled the code first to review instead of just relying on the github interface.